src/scipp/core/reduction.py

```diff
@@ -83,6 +97,481 @@ def nanmean(x: VariableLikeType, dim: Optional[str] = None) -> VariableLikeType:
+    along each reduced dimension.
+    That is, for an array of ``N`` unmasked, non-NaN values, the median is
+
+    - odd ``N``: ``x[(N-1)/2]``
+    - even ``N``: ``(x[N/2-1] + x[N/2]) / 2``
+
+    Parameters
+    ----------
+    x: scipp.typing.VariableLike
+        Input data.
+    dim:
+        Dimension(s) along which to calculate the median.
+        If not given, the median over a flattened version of the array is calculated.
+
+    Returns
+    -------
+    : Same type as x
+        The median of the input values.
+
+    Raises
+    ------
+    scipp.VariancesError
+        If the input has variances.
+    ValueError
+        If the input has masks.
+        Mask out NaN's and then use :func:`scipp.median` instead.
+
+    See Also
+    --------
+    scipp.median:
+        Compute the median without special handling of NaN's.
+
+    Examples
+    --------
+    ``nanmedian`` is available as a method:
+
+        >>> x = sc.array(dims=['x'], values=[2, 5, 1, np.nan, 8, 4])
+        >>> x.nanmedian()
+        <scipp.Variable> ()    float64  [dimensionless]  4
+        >>> x = sc.array(dims=['x'], values=[2, np.nan, 5, 1, 8])
+        >>> x.nanmedian()
+        <scipp.Variable> ()    float64  [dimensionless]  3.5
+    """
+
+    def _catch_masked(*args, **kwargs):
+        # Because there is no np.ma.nanmedian
+        raise ValueError(
+            'nanmedian does not support masked data arrays. '
+            'Consider masking NaN values and calling scipp.median'
+        )
+
+    return _reduce_with_numpy(
+        x,
+        dim=dim,
+        sc_func=nanmedian,
+        np_func=np.nanmedian,
+        np_ma_func=_catch_masked,
+        unit_func=lambda u: u,
+        kwargs={},
+    )
+
+
+def var(x: VariableLikeType, dim: Dims = None, *, ddof: int) -> VariableLikeType:
+    r"""Compute the variance of the input values.
+
+    This function computes the variance of the input values which is *not*
+    the same as the ``x.variances`` property but instead defined as
+
+    .. math::
+
+        \mathsf{var}(x) = \frac1{N - \mathsf{ddof}}
+                          \sum_{i=1}^{N}\, {(x_i - \bar{x})}^2
+
+    where :math:`x_i` are the unmasked ``values`` of the input and
+    :math:`\bar{x}` is the mean, see :func:`scipp.mean`.
+    See the ``ddof`` parameter description for what value to choose.
+
+    Note
+    ----
+    Masks are broadcast to the shape of ``x``.
+    This can lead to a large temporary memory usage.
+
+    Parameters
+    ----------
+    x: scipp.typing.VariableLike
+        Input data.
+    dim:
+        Dimension(s) along which to calculate the variance.
+        If not given, the variance over a flattened version of the array is calculated.
+    ddof:
+        'Delta degrees of freedom'.
+        For sample variances, set ``ddof=1`` to obtain an unbiased estimator.
+        For normally distributed variables, set ``ddof=0`` to obtain a maximum
+        likelihood estimate.
+        See :func:`numpy.var` for more details.
```

Comment:
Do you want to add a comment (or even a specific error?) why this is required by Scipp? Otherwise we risk a future developer coming and changing this.